### PR TITLE
Missing api calls

### PR DESCRIPTION
**Pull request by Bogdan-D**
# Submit a pull request

## CLA

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [x] The code changes follow best practices
- [ ] Code changes are tested (add some information if not applicable)

## Description of the pull request
Implementation of missing methods:
1. GetApp, UpdateAppSettings
2. UpdateUser
3. ReactivateUser
4. GetMessage
5. Flag\Unflag message
6. UpdateChannelType
7. VerifyWebhook
8. channel.Query
9. Search

### CODE REVIEW

File: app.go [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+	}
+
+	return resp.App, nil
+}
+
+// UpdateAppSettings makes request to update app settings
+// Example of usage:
+//  settings := NewAppSettings().SetDisableAuth(true)
+//  err := client.UpdateAppSettings(settings)
+func (c *Client) UpdateAppSettings(settings *AppSettings) error {
```
Comment (pascaldekloe) [inline app.go]: Update means you replace the settings with the provided object. With all that pointer use, I assume you're sending a patch here?

```go
// PatchAppConfig requests field changes for each non nil value.
func (c *Client) PatchAppConfig(patch  *AppConfigPatch) error {
```

File: query_test.go [hunk 2]
```diff
@@ -40,3 +40,29 @@ func TestClient_QueryChannels(t *testing.T) {
...
+func TestClient_Search(t *testing.T) {
+	c := initClient(t)
+	ch := initChannel(t, c)
+
+	user1, user2 := randomUser(), randomUser()
+
+	text := randomString(10)
+
+	msg1, err := ch.SendMessage(&Message{Text: text + " " + randomString(25)}, user1.ID)
+	mustNoError(t, err)
```
Comment (pascaldekloe) [inline query_test.go]: Labels! 😡

File: app.go [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+	DisableAuth          bool                      `json:"disable_auth_checks"`
+	DisablePermissions   bool                      `json:"disable_permissions_checks"`
+}
+
+type appResponse struct {
+	App *AppResponseFields `json:"app"`
+}
+
+// GetApp returns app settings
+func (c *Client) GetApp() (*AppResponseFields, error) {
```
[the remark this answers is not in the document]
Comment (Bogdan-D) [reply app.go]: Don't agree, method names should be about the same on all SDK's

File: app.go (hunk 1 above)
Comment (Bogdan-D) [reply app.go]: same as above

File: app.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (Bogdan-D) [reply app.go]: I'd renamed AppResponseFields -> AppConfig; I think good solution woulg be to name it as `GetAppConfig()`

[the remark this answers is not in the document]
Comment (pascaldekloe) [reply app.go]: In terms of documentation that makes sense.

File: app.go (hunk 1 above)
Comment (pascaldekloe) [reply app.go]: Then document the behaviour.

**Pull request merged**